library/std/src/net/ip/display_buffer.rs

```diff
@@ -0,0 +1,40 @@
+use crate::fmt;
+use crate::mem::MaybeUninit;
+use crate::str;
+
+/// Used for slow path in `Display` implementations when alignment is required.
+pub struct IpDisplayBuffer<const SIZE: usize> {
+    buf: [MaybeUninit<u8>; SIZE],
+    len: usize,
+}
+
+impl<const SIZE: usize> IpDisplayBuffer<SIZE> {
+    #[inline(always)]
+    pub const fn new(_ip: &[u8; SIZE]) -> Self {
```

Comment:
This constructor is pretty idiosyncratic. I don't know that I think it's a good way of doing it.

I guess it's fine for now, but I might rename the function name to `with_capacity_for` or something, to make it clear that the argument is mostly unused and is just provided as a way to ensure capacity.

Reply:
Removed the argument.